packages/wp-dashboard/src/components/editorSettings/customFonts/index.js

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useState, useCallback, useEffect } from '@web-stories-wp/react';
+import { __ } from '@web-stories-wp/i18n';
+import {
+  Button,
+  BUTTON_SIZES,
+  BUTTON_TYPES,
+  BUTTON_VARIANTS,
+  Icons,
+  Text,
+  THEME_CONSTANTS,
+} from '@web-stories-wp/design-system';
+import styled from 'styled-components';
+import getFontDataFromUrl from '@web-stories-wp/fonts/src/utils/getFontDataFromUrl';
+// @todo Replace.
+import { isValidUrl } from '@web-stories-wp/story-editor/src/utils/url';
+
+/**
+ * Internal dependencies
+ */
+import Tooltip from '@web-stories-wp/story-editor/src/components/tooltip';
+import {
+  InlineForm,
+  SaveButton,
+  SettingForm,
+  SettingHeading,
+  SettingsTextInput,
+  SettingSubheading,
+  TextInputHelperText,
+} from '../components';
+import ConfirmationDialog from './confirmationDialog';
+
+export const TEXT = {
+  ADD_CONTEXT: __(
+    'Add and manage your custom and brand fonts via link here. They’ll appear in your font list in the editor.',
+    'web-stories'
+  ),
+  REMOVAL: __(
+    'Removing a font from this settings page will remove it from your font list in all stories.',
+    'web-stories'
+  ),
+  SECTION_HEADING: __('Custom Fonts', 'web-stories'),
+  LABEL: __('Insert Font URL', 'web-stories'),
+  INPUT_CONTEXT: __(
+    'Insert a public URL to a font file. Allowed formats are .otf, .ttf, .woff.',
+    'web-stories'
+  ),
+  INPUT_ERROR: __('Invalid URL format', 'web-stories'),
+  SUBMIT_BUTTON: __('Add Font', 'web-stories'),
+  FONTS_HEADING: __('Current Fonts', 'web-stories'),
+  FONTS_CONTEXT: __(
+    'Deleting fonts will delete them from your in-editor font list and all previous stories.',
+    'web-stories'
+  ),
+};
+
+const AddButton = styled(SaveButton)`
+  min-width: 90px;
+  margin-top: 32px;
+`;
+
+const InputsWrapper = styled.div`
+  margin-top: 8px;
+`;
+
+const FontsWrapper = styled.div`
+  margin-top: 34px;
+`;
+
+const ListHeading = styled(Text)`
+  margin-bottom: 10px;
+  display: inline-block;
+`;
```

Comment:
it's a bummer to me that you're having to add margin in to the customFonts specifically - just on first glance. When this is up for review I'll look closer, but there should be reusable StyledComponents in `../components` to keep every section as its own form.

Reply:
I'll take another look, couldn't find a component currently with the required margin but maybe I could still reuse one with `inline-block` and override the margin then, if that makes more sense.

Reply:
i wouldn't worry too much about it, i'll look at this when i pull the code to review.